scripts/core/editor3/components/Editor3.jsx

```diff
@@ -20,146 +21,137 @@ import Toolbar from './toolbar';
  * @description Editor3 is a draft.js based editor that support customizable
  *  formatting (shortly more features will come ...)
  */
-export class Editor3 extends React.Component {
-    constructor(props) {
-        super(props);
 
-        const initialContentState = stateFromHTML(props.value);
+export const getInitialState = (spellcheckerService, value, onChangeValue, readOnly, showToolbar, editorFormat) => {
+    const initialContentState = stateFromHTML(value);
 
-        this.state = {
-            editorState: EditorState.createWithContent(
+    return {
+        editorState: EditorState.createWithContent(
                 initialContentState,
-                this.getDecorators()
-            )
-        };
+                getDecorators(spellcheckerService)
+        ),
+        spellcheckerState: {spellcheckerService: spellcheckerService},
```

Comment:
For editor state the name in DraftJs is editorState, so I used it and I used the same pattern for spellchecker to be consistent in naming things. But yes, if is needed  it can be changed later when we add other modules and use combineReducers to combine reducers. In this case every reducer uses a key to access state. If the key is editor3, the store tree will be state.editor3.editorState and state.editor3.spellcheckerState.